src/material/schematics/ng-update/upgrade-rules/hammer-gestures-v9/hammer-gestures-rule.ts

```diff
@@ -134,7 +134,51 @@ export class HammerGesturesRule extends MigrationRule<null> {
     const hasCustomGestureConfigSetup =
         this._hammerConfigTokenReferences.some(r => this._checkForCustomGestureConfigSetup(r));
 
-    if (this._usedInRuntime || this._usedInTemplate) {
+    /*
+      Possible scenarios and how the migration should change the project:
+        1. We detect that a custom HammerJS gesture config set up:
+            - Remove references to the Material gesture config if no event from the
+              Angular Material gesture config is used.
+            - Print a warning about ambiguous setup that cannot be handled completely.
+        2. We detect that HammerJS is *only* used at runtime:
+            - Remove references to GestureConfig of Material.
+            - Remove references to the "HammerModule" if present.
+        3. We detect that HammerJS is used in a template:
+            - Copy the Material gesture config into the app.
+            - Rewrite all gesture config references to the newly copied one.
+            - Setup the new gesture config in the root app module.
+        4. We detect no HammerJS usage at all:
+            - Remove Hammer imports
+            - Remove Material gesture config references
+            - Remove HammerModule setup if present.
+            - Remove Hammer script imports in "index.html" files.
+    */
+
+    if (hasCustomGestureConfigSetup) {
+      // If a custom gesture config is provided, we always assume that HammerJS is used.
+      HammerGesturesRule.globalUsesHammer = true;
+      if (!this._usedInTemplate && this._gestureConfigReferences.length) {
+        // If the Angular Material gesture events are not used and we found a custom
+        // gesture config, we can safely remove references to the Material gesture config
+        // since events provided by the Material gesture config are guaranteed to be unused.
+        this._removeMaterialGestureConfigSetup();
+        this.printInfo(
+          'The HammerJS v9 migration for Angular components detected that HammerJS is ' +
+          'manually set up in combination with references to the Angular Material gesture ' +
+          'config. The migration is unable to perform the full migration for this target, ' +
```

Comment:
"The migration is unable to perform the full migration" sounds kind of weird, how about: "This target cannot be fully migrated, but ..."

Reply:
Agreed. should be better now!